crates/core/src/factories/mem_transport.rs

```diff
@@ -349,80 +428,128 @@ impl Listener {
-        if let Some(send) = map.get(to_peer) {
-            return Some(send.send.clone());
+        if let Some(open_connection) = conn_pool.get(to_peer) {
+            return Some(open_connection.data_send.clone());
         }
 
-        let id: u64 = match to_peer.peer_id() {
+        let to_peer_id: u64 = match to_peer.peer_id() {
             None => return None,
             Some(id) => match id.parse() {
                 Err(_) => return None,
                 Ok(id) => id,
             },
         };
 
-        let send = match self.con_map.lock().unwrap().get(&id) {
-            None => return None,
-            Some(send) => send.clone(),
-        };
-
-        let (ds1, dr1) = tokio::sync::mpsc::unbounded_channel();
-        let (ds2, dr2) = tokio::sync::mpsc::unbounded_channel();
-
-        if send.send((from_peer.clone(), ds1, dr2)).is_err() {
+        // Get the sender side of the connection establishment channel for the
+        // MemTranport associated with the to_peer Url, if an associated
+        // MemTransport instance is found.
+        let connection_send =
+            match self.instances_map.lock().unwrap().get(&to_peer_id) {
+                None => return None,
+                Some(send) => send.clone(),
+            };
+
+        // Create an unbounded "data" channel "B -> A" to send data from
+        // peer B's MemTransport instance to peer A's MemTransport instance
+        let (data_send_1, data_recv_1) = tokio::sync::mpsc::unbounded_channel();
+
+        // Create an unbounded "data" channel "A -> B" to send data from
+        // peer A's MemTransport instance to peer B's MemTransport instance
+        let (data_send_2, data_recv_2) = tokio::sync::mpsc::unbounded_channel();
+
+        // Send the sending end of the "data" channel "B > A" and the receiving
+        // end of the "data" channel "A > B" to the MemTransport instance
+        // of peer B.
+        if connection_send
+            .send((from_peer.clone(), data_send_1, data_recv_2))
+            .is_err()
+        {
             return None;
         }
 
-        let _ = cmd_send.send(Cmd::RegCon(to_peer.clone(), ds2.clone(), dr1));
+        // Send the sending end of the "data" channel "A > B" and the receiving
+        // end of the "data" channel "B > A" to the MemTransport instance
+        // of peer A.
+        let _ = cmd_send.send(Cmd::RegisterConnection(
+            to_peer.clone(),
+            data_send_2.clone(),
+            data_recv_1,
+        ));
 
-        Some(ds2)
+        Some(data_send_2)
     }
 }
 
-/// This is our static global instance of the [Stat] struct.
-static STAT: OnceLock<Stat> = OnceLock::new();
-fn get_stat() -> &'static Stat {
-    STAT.get_or_init(Stat::new)
+/// This is our static global instance of the [`TRANSPORT_INSTANCES`] struct.
+static TRANSPORT_INSTANCES: OnceLock<TransportInstances> = OnceLock::new();
+/// Get the global instance of the TransportInstances struct which holds the sender side of
```

Comment:
```suggestion
/// Get the global instance of the [`TransportInstances`] struct which holds the sender side of
```

This can link I think?